scylla-cql/src/macros.rs

```diff
@@ -75,6 +75,68 @@ pub use scylla_macros::ValueList;
 /// to either the `scylla` or `scylla-cql` crate.
 pub use scylla_macros::SerializeCql;
 
+/// Derive macro for the [`SerializeRow`](crate::types::serialize::row::SerializeRow) trait
+/// which serializes given Rust structure into bind markers for a CQL statement.
+///
+/// At the moment, only structs with named fields are supported. The generated
+/// implementation of the trait will match the struct fields to bind markers/columns
+/// by name automatically.
+///
+/// Serialization will fail if there are some bind markers/columns in the statement
+/// that don't match to any of the Rust struct fields, _or vice versa_.
+///
+/// In case of failure, either [`BuiltinTypeCheckError`](crate::types::serialize::row::BuiltinTypeCheckError)
+/// or [`BuiltinSerializationError`](crate::types::serialize::row::BuiltinSerializationError)
+/// will be returned.
+///
+/// # Example
+///
+/// A UDT defined like this:
+///
+/// ```notrust
+/// CREATE TYPE ks.my_udt (a int, b text, c blob);
+/// ```
+///
+/// ...can be serialized using the following struct:
+///
+/// ```rust
+/// # use scylla_cql::macros::SerializeRow;
+/// #[derive(SerializeRow)]
+/// # #[scylla(crate = scylla_cql)]
+/// struct MyUdt {
+///     a: i32,
+///     b: Option<String>,
+///     c: Vec<u8>,
+/// }
+/// ```
```

Comment:
This is probably not the example you meant to put here, it should be one of statement with bind markers / columns and corresponding struct, no?

Reply:
Yes. It was copied from the SerializeCql example but I didn't update the description and the CQL part. I'll fix.